### PR TITLE
[CSL-2209] Initial draft of mutually auth between server-client

### PR DESCRIPTION
**Pull request by akegalj**

### CODE REVIEW

File: wallet-new/docs/wallet-tls.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+# Abstract<a id="sec-2" name="sec-2"></a>
+
+The purpose of this document is to review security model in communication
+ between wallet backend and wallet frontend (Daedalus).
+
+# Motivation<a id="sec-3" name="sec-3"></a>
+
+Communication between wallet frontend (Daedalus) and wallet backend in cardano
+ is done via http server. There are various attacks possible to this kind of
```
Comment (adinapoli-iohk) [inline wallet-new/docs/wallet-tls.md]: Wording and grammar 😉

File: wallet-new/docs/wallet-tls.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+ is done via http server. There are various attacks possible to this kind of 
+environmnet (TODO footnote or further list of possible attack vectors needed).
+
+# Current design<a id="sec-4" name="sec-4"></a>
+
+This summarises current approach in server-client communication.
+
+-   wallet server has https server (based on TLS CA server cert) set up and listening on specific IP:PORT
+    
+-   client is using its on pair of CA certificate in order to verify server cert when sending request to server
```
Comment (adinapoli-iohk) [inline wallet-new/docs/wallet-tls.md]: _its own pair_, maybe?

Comment (adinapoli-iohk) [reply wallet-new/docs/wallet-tls.md]: Where is the evidence (with hypelinks to actual code) to back those assertions up? I guess that's arguably the most useful aspect of a technical proposal: bridge the gaps between the theory and the reality of the current technical implementation 😉

File: wallet-new/docs/wallet-tls.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+-   the previous scheme, partly based on TLS CA server cert (provided by the backend) and the authentication password (provided by the frontend, CSM-100) was deemed too be unnecessarily hostile to the user (who had to remember Yet Another Password)
+
+-   previous point means we have to make do with a non-interactive mutual authentication scheme
+
+-   PKI provides us with such a scheme – the CA might be used to sign both the backend key/certificate pair (which we already do in DEVOPS-33/CSM-116) and the frontend key/certificate pair
+
+
+The proposal is to implement this scheme:
+
+-   Instead of three artifacts (CA cert, backend key, backend cert), the installer will generate five (+frontend key, +frontend cert) - already implemented
```
Comment (adinapoli-iohk) [inline wallet-new/docs/wallet-tls.md]: Do you have at hand the link to the PR which ships this? I guess it's one of those things really useful to see, as a "witness".

File: wallet-new/docs/wallet-tls.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+-   The frontend will have to provide the cert to the backend
+
+-   The backend will switch from ignoring the client certificate to requiring a valid one to be provided
+
+## Nodejs frontend change<a id="sec-5-1" name="sec-5-1"></a>
+
+Current implementation (29 Jan 2018) is using nodejs `https` library which exposes `ca` placeholder for client side TLS authentication [daedalus request](https://github.com/input-output-hk/daedalus/blob/eaec35f44bbfe4335445d635a53fbbcb3b2e686c/app/api/etc/lib/request.js#L4) :
```
Comment (adinapoli-iohk) [inline wallet-new/docs/wallet-tls.md]: Good, exactly like you did here (in terms of linking actual code)! 😉

**Comment by akegalj** [comment]
Closing as it seems it won't be continued. Reopen if you find this useful.

**Pull request closed without merging**